### PR TITLE
feat: line scatter chart

### PR DESCRIPTION
**Pull request by jung-han**
### Please check if the PR fulfills these requirements
- [ ] It's submitted to right branch according to our branching model
- [ ] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

#### options

```ts
const options = {
  series: {
    line: {
      showDot: boolean;
      lineWidth: number;
      spline: boolean;
    },
    dataLabels: DataLabels;
    selectable: boolean; 
  }
}
```

#### screen shot

image

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Review by lja1018 — APPROVED** [review]
[08/27] 리뷰완료.
고생하셨습니다!!

**Review by shiren — APPROVED** [review]
수고하셨어요

File: src/component/lineSeries.ts [hunk 1]
```diff
@@ -154,12 +172,14 @@ export default class LineSeries extends Component {
...
-    limit: ValueEdge,
+    scale: Scale,
     renderOptions: RenderOptions,
     categories: string[]
   ): LinePointsModel[] {
-    const { pointOnColumn, options, tickDistance } = renderOptions;
+    const { pointOnColumn, options, tickDistance, labelDistance } = renderOptions;
     const { spline, lineWidth } = options;
+    const yAxisLimit = scale.yAxis.limit;
+    const xAxisLimit = scale?.xAxis?.limit;
```
Comment (jungeun-cho) [inline src/component/lineSeries.ts:182]: 여기 옵셔널체이닝이 필요한거에요?

File: src/component/tooltip.ts [hunk 2]
```diff
@@ -78,49 +84,51 @@ export default class Tooltip extends Component {
...
-          acc.templateType = templateType;
-        }
-
-        return acc;
-      },
-      { type: 'tooltip', x: 0, y: 0, data: [], target: { radius: 0, width: 0, height: 0 } }
-    );
+  renderTooltip() {
+    const model = Object.values(this.tooltipInfoModels)
+      .flatMap((item) => item)
```
Comment (jungeun-cho) [inline src/component/tooltip.ts:89]: 확장해서 가져오는 models만 가져오는 부분 함수로 따로 분리하는건어때요?
`this.getModels()` 또는 `this.flatModels()` 로요.
```ts
this.getModels()
  .redude...

getModels() {
  return Object.values(this.tooltipInfoModels).flatMap((model) => model);
}
```

File: src/component/lineSeries.ts [hunk 3]
```diff
@@ -62,14 +72,22 @@ export default class LineSeries extends Component {
       throw new Error("There's no line data!");
     }
 
-    const { yAxis } = scale;
-    const { tickDistance, pointOnColumn } = axes.xAxis!;
+    const options = { ...chartState.options };
+    if ((options.series as LineScatterChartSeriesOptions)?.line) {
```
[the remark this answers is not in the document]
Comment (jung-han) [reply src/component/lineSeries.ts:76]: 이 조건문에서 `LineScatterChartSeriesOptions` 타입의 옵션을 `LineChartOptions`로 항상 만드는게 목표여서요!

```ts
if (options?.series && 'line' in options.series) {
  options.series = { ...options.series, ...options.series.line };
}
```

로 수정하겠습니다!

**Review by jungeun-cho — APPROVED** [review]
[8/27] 리뷰 완료합니다. 수고하셨어요. :+1:

File: src/component/lineSeries.ts (hunk 1 above)
Comment (jung-han) [reply src/component/lineSeries.ts:182]: x축의 limit은 coordinate차트만 생기게 됩니다!

File: src/component/tooltip.ts (hunk 2 above)
Comment (jung-han) [reply src/component/tooltip.ts:89]: 좋습니다!

**Pull request merged**